### ISSUE TITLE
.Net: Function calling to call only one function at a time (parallel_tool_calls = false)

### ISSUE THREAD

**Issue by dcostea** [microsoft/semantic-kernel#6636]
I'm using function calling to call a sequence of native functions. The exact sequence is important but the implemented behavior is the default behavior (if I'm not wrong) and this is implementing _parallel_tool_calls = true_.

I'm wondering if we can have this property in OpenAIPromptExecutionSettings class. Ideally something like this:

```
OpenAIPromptExecutionSettings
{
  ParallelToolCalls = false,
  ToolCallBehavior = ...
}
```

**Comment by stephentoub** [comment]
> the implemented behavior is the default behavior (if I'm not wrong) and this is implementing parallel_tool_calls = true.

You're correct.

cc: @SergeyMenshykh, for clarity, this is referring to:
https://platform.openai.com/docs/guides/function-calling
"Parallel function calling can be disabled by passing parallel_tool_calls: false in the request. The model will only call one function at a time when parallel function calling is disabled."

**Comment by macmanad** [comment]
I have encountered an issue where _it seems_ i am giving OpenAI a number of similar functions, and it chokes while trying to call them (500 response... mostly...). When curling, if i pass `parallel_tool_calls = false`, it responds fine with the first function call. I went looking for a way to pass this through SK, and found this ticket saying it's not available yet, so kept looking.

I then came across [Extension Data](https://learn.microsoft.com/en-us/dotnet/api/microsoft.semantickernel.promptexecutionsettings.extensiondata?view=semantic-kernel-dotnet#microsoft-semantickernel-promptexecutionsettings-extensiondata) that at least as how i read the docco, indicates that it allows exactly what's wanted, arbitrary other kvps to be passed along to the llm. 

> Extra properties that may be included in the serialized execution settings


Except it doesn't. 
[CreateChatCompletionsOptions](https://github.com/microsoft/semantic-kernel/blob/6b59077ff7c7df71e9d7b4986a2e2350b0e231a2/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L1056) does not copy it into the ChatCompletionsOptions, which are what gets passed into the [serialization for the request](https://github.com/microsoft/semantic-kernel/blob/6b59077ff7c7df71e9d7b4986a2e2350b0e231a2/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L653). It only seems to actually be used to [add diagnostic tags ](https://github.com/microsoft/semantic-kernel/blob/6b59077ff7c7df71e9d7b4986a2e2350b0e231a2/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L647).

Did i entirely misinterpret the docco?
